network.go

```diff
@@ -25,10 +25,10 @@ import (
 // related information.
 type network struct {
 	ifacesLock sync.Mutex
-	ifaces     []*pb.Interface
+	ifaces     []pb.Interface
```

Comment:
As I mentioned in https://github.com/kata-containers/agent/pull/225/files#r184835917, we can use pb.Interface.Name to uniquely identify an interface and thus use a map instead of a slice to store it. Then it is perfectly safe to call delete in a loop.

Reply:
Either way is fine with me. Do you really want that we introduce maps here, or you're okay with this way ?

Reply:
Please change to use a map for it. It has a few advantages:
- suitable for delete within a loop
- ensuring each interface is unique

Reply:
Ok, it works in this case because the name is already what we use to identify what we need to remove.